addons/survey/static/src/views/fields/boolean_toggle/survey_boolean_toggle_field.js

```diff
@@ -0,0 +1,14 @@
+/** @odoo-module **/
+import { registry } from "@web/core/registry"
+import { BooleanToggleField } from '@web/views/fields/boolean_toggle/boolean_toggle_field';
+
+export class SurveyBooleanToggleField extends BooleanToggleField {
+    /**
+    *override
+    */
+    onChange(newValue) {
+        this.props.update(newValue);
```

Comment:
I'm genuinely unaware of this, is this good practice vs. updating the `props.record` like in the original component?

Reply:
Sorry I have to insist here. Looking for why we don't see more like this, I found this PR which actively removed `props.update` calls: #112792.

Reply:
Please check out the changes!!